homeassistant/components/nextdns/select.py

```diff
@@ -0,0 +1,164 @@
+"""Support for the NextDNS service."""
+from __future__ import annotations
+
+import asyncio
+from collections.abc import Callable
+from dataclasses import dataclass
+from typing import Any, Generic
+
+from aiohttp import ClientError
+from aiohttp.client_exceptions import ClientConnectorError
+from nextdns import ApiError, Settings
+
+from homeassistant.components.select import SelectEntity, SelectEntityDescription
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.entity import EntityCategory
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import CoordinatorDataT, NextDnsSettingsUpdateCoordinator
+from .const import ATTR_SETTINGS, DOMAIN
+
+PARALLEL_UPDATES = 1
+
+LOCATION_MAP = {
+    "switzerland": "ch",
+    "european_union": "eu",
+    "great_britain": "gb",
+    "united_states": "us",
+}
+LOCATION_INVERTED_MAP = {v: k for k, v in LOCATION_MAP.items()}
+
+RETENTION_MAP = {
+    "one_hour": 1,
+    "six_hours": 6,
+    "one_day": 24,
+    "one_week": 168,
+    "one_month": 720,
+    "three_months": 2160,
+    "six_months": 4320,
+    "one_year": 8760,
+    "two_years": 17520,
+}
+RETENTION_INVERTED_MAP = {v: k for k, v in RETENTION_MAP.items()}
+
+
+@dataclass
+class NextDnsSelectRequiredKeysMixin(Generic[CoordinatorDataT]):
+    """Class for NextDNS entity required keys."""
+
+    current_option: Callable[[CoordinatorDataT], str]
+    select_option_method: str
+    option_map: dict[str, Any]
+
+
+@dataclass
+class NextDnsSelectEntityDescription(
+    SelectEntityDescription, NextDnsSelectRequiredKeysMixin[CoordinatorDataT]
+):
+    """NextDNS select entity description."""
+
+
+SELECTS = (
+    NextDnsSelectEntityDescription[Settings](
+        key="logs_location",
+        name="Logs location",
```

Comment:
Why would we want to have these options in HA. Is there any reason someone would want to automate this ?

Reply:
I had a question from the user whether all NextDNS configuration options will be available in HA, he wanted to move the entire NextDNS configuration panel to HA. Frankly, I also use integration in this way, if I need to change something in NextDNS, I do it through HA. As for the `logs location` itself, I don't know how someone might want to automate it. Do you think to remove it? Maybe the entity should be disabled by default?

Reply:
I would just leave this one out. No one will ever need it. HA aim is to be an automation platform, not a config panel for every service in the world.

Reply:
`logs_location` removed.